## Loading library

In [1]:
from xiespp import formation_energy

## Loading data from Materials Project
The input to the model can in formats like: CIF, POSCAR, PyMatGen Structure, ASE Atoms

In [6]:
from mp_api.client import MPRester

api_key = '__YOUR_API_KEY__'
mpr = MPRester(api_key)

search = mpr.summary.search(chemsys="Si-O")
data_input = search[:10]

Retrieving SummaryDoc documents:   0%|          | 0/344 [00:00<?, ?it/s]

## Preparing data for passing to the model

In [7]:
gen = formation_energy.prepare_data(
    data_input=data_input, 
    # input_format= 'vasp', # Use this if the format of the file is not CIF file
)

Preparing image objects...
Checking image requirements...


100%|██████████| 10/10 [00:00<00:00, 72.77it/s]


Valid images:       8 /      10
Preparing point clouds...


100%|██████████| 8/8 [00:00<00:00, 2647.50it/s]


In [8]:
# Some crystals did not meet the condition of the formation energy calculation.
gen.df

atoms  ...  is_valid
0  MPDataDoc<SummaryDoc>\n[1...  ...      True
1  MPDataDoc<SummaryDoc>\n[1...  ...      True
2  MPDataDoc<SummaryDoc>\n[1...  ...      True
3  MPDataDoc<SummaryDoc>\n[1...  ...      True
4  MPDataDoc<SummaryDoc>\n[1...  ...     False
5  MPDataDoc<SummaryDoc>\n[1...  ...      True
6  MPDataDoc<SummaryDoc>\n[1...  ...      True
7  MPDataDoc<SummaryDoc>\n[1...  ...      True
8  MPDataDoc<SummaryDoc>\n[1...  ...      True
9  MPDataDoc<SummaryDoc>\n[1...  ...     False

[10 rows x 4 columns]

## Making the predictions by the model

In [9]:
model = formation_energy.FormationEnergyPredictor()
model

Metal device set to: Apple M1

systemMemory: 16.00 GB
maxCacheSize: 5.33 GB



<FormationEnergyPredictor: device=/device:CPU:0, ensemble=50>

In [10]:
yp = model.predict(gen, return_all_ensembles=True)

50/50 [==============================] - 30s 596ms/step


In [11]:
# Besides the final prediction (mean), all the single values of ensembles are being reported:
# The missing values are the ones that did not meet the criteria of the formation energy model
# The mean value is the final prediction of the model. 
# If return_all_ensembles is set to False, only the mean value is returned.
yp

mean         0         1  ...        47        48        49
0 -2.167284 -2.164894 -2.101233  ... -2.201219 -2.280266 -2.135353
1 -3.253676 -3.295523 -3.218916  ... -3.271765 -3.251513 -3.274720
2 -3.221661 -3.250412 -3.189190  ... -3.221860 -3.227841 -3.189372
3 -3.181098 -3.166312 -3.184871  ... -3.152972 -3.270755 -3.219997
4       NaN       NaN       NaN  ...       NaN       NaN       NaN
5 -3.252608 -3.313522 -3.259411  ... -3.203030 -3.230890 -3.296526
6 -0.562573 -0.607019 -0.544663  ... -0.685055 -0.624325 -0.609955
7 -3.208578 -3.062743 -3.264846  ... -3.173977 -3.245857 -3.212483
8 -3.198319 -3.177821 -3.186000  ... -3.223092 -3.218949 -3.176019
9       NaN       NaN       NaN  ...       NaN       NaN       NaN

[10 rows x 51 columns]

## Command line usage:
~~~sh
(syn) ➜  ~ xiespp_formation_energy --test -v
~~~
Results:
~~~
Running on test samples: ['/Users/ali/mambaforge/envs/syn/lib/python3.8/site-packages/xiespp-1.0.700-py3.8.egg/xiespp/synthesizability_1/data/GaN/GaN_9.cif', '/Users/ali/mambaforge/envs/syn/lib/python3.8/site-packages/xiespp-1.0.700-py3.8.egg/xiespp/synthesizability_1/data/GaN/GaN_12.cif']
Metal device set to: Apple M1

systemMemory: 16.00 GB
maxCacheSize: 5.33 GB

Reading CIF files...
100%|█████████████████████████████████████████████| 2/2 [00:00<00:00, 46.16it/s]
Preparing image objects...
Checking image requirements...
100%|████████████████████████████████████████████| 2/2 [00:00<00:00, 328.30it/s]
Valid images:       2 /       2
Preparing point clouds...
100%|███████████████████████████████████████████| 2/2 [00:00<00:00, 1349.52it/s]
50/50 [==============================] - 8s 148ms/step
[-0.5372748589515686, 0.25509491860866546]
~~~

~~~sh
(syn) ➜  ~ xiespp_formation_energy -f Downloads/LiF.POSCAR --format vasp
~~~
Results:
~~~
[-3.127022614479065]
~~~